optimum/habana/transformers/trainer.py

```diff
@@ -986,7 +986,10 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args, use_reentrant: Optio
                             inputs["flash_attention_causal_mask"] = True
                 if self.model.config is not None:
                     if self.model.config.model_type in ["llama", "qwen2", "mistral", "starcoder2"]:
-                        inputs["lazy_mode"] = args.use_lazy_mode
+                        forward_method = getattr(self.model, "forward")
+                        signature = inspect.signature(forward_method)
+                        if "lazy_mode" in signature.parameters:
```

Comment:
one case this doesnt cover is peft as model class is PeftModelForCausalLM- checking to see if there's a way around that